Documentation/General/productPlanning/requirements.tex

```diff
@@ -0,0 +1,146 @@
+		\item Nervous
+		\item Exiting
+		\item Happy
+		\item Pleasing
+		\item Relaxing
+		\item Peaceful
+		\item Calm
+		\item Sleepy
+		\item Bored
+		\item Sad
+		\end{itemize}
+	  These moods are defined by the Arousal/Valence model\footnote{Y. Chang and H.H. Chen. Music Emotion Recognition. Boca raton, State:
+CRC Press, 2011.}.
+
+\item The system shall serve the user a list of rooms, ordered by how close they match the given mood.
+
+\item The web interface shall present the user with a room when it is chosen.
+	
+\item The system shall stream the music from the SoundCloud platform using the SoundCloud API\footnote{\url{https://developers.soundcloud.com/docs/api/guide}}
+
+\item The system shall use a multi-layer perceptron system to predict the mood of a song.
+
+\item The system will use a recommender system to determine the next song to be played in each room.
+
+\end{itemize}
+\subsubsection{Should haves}
+\begin{itemize}
+\item The web interface shall start playing the current song on the current timestamp when a user joins the room.
+
+\item A room shall contain a chat, connecting people with the same mood together.
+
+\item A room shall contain a now playing slider.
+
+\item The web interface shall give the user the option to mute the playing music.
+
+\item The web interface shall notify the user if they made an illogical combination of moods, warning them results may vary.
+
+\item A room has a button to upvote the current song, indicating the song is a good fit to the room.
+
+\item A room has a button to downvote the current song, indicating the song is a bad fit to the room. This will also vote to skip.
+
+\item A room has a button to "vote to skip". This does not affect the fit of the song to the room, but is a tool to skip songs.
+
+\item A song shall be skipped in one of the following situations:
+	\begin{itemize}
+	\item The song received $\frac{\$number\_of\_users\_in\_room}{3}$ votes to skip.
+	\item The song received at least 10 votes, and the ratio upvote:downvote is 1:2 or less.
+	\item A moderator skips the song (to prevent escalations)
+	\end{itemize}
+	
+\item The system shall periodically suggest the user to hop to another room to encourage more interaction and we can direct the user to a more preferred mood.
+
+\item The multi-layer perceptron shall be trained with metadata provided by EEMCS
+
+\item The upvotes and downvotes for a song change the weights of the multi-layer perceptron.
+
+\item The user shall be rewarded for up- and downvoting with "treats". These treats are achievement points.
+
+\item The system shall retrieve more metadata about a song from AcousticBrainz\footnote{\url{http://acousticbrainz.org/}}.
+
+\item If metadata is not present in the AcousticBrainz database the system shall run an analysis itself using the AcousticBrainz software.
+
+\end{itemize}
+
+\subsubsection{Could haves}
+\begin{itemize}
+\item The system shall provide a search engine that aids the user in finding a song to add.
```

Comment:
This would also require a playing queue rather than generating a next song to be played.